packages/render-fidelity-tools/src/artifact-creator.ts

```diff
@@ -250,29 +250,30 @@ export class ArtifactCreator {
     // variables are captured in its closure scope. TypeScript compiler
     // currently has no mechanism to detect this and will happily tell you
     // your code is correct when it isn't.
-    const evaluateError =
-        await page.evaluate(async (useMaxTime, maxTimeInSec) => {
-          const modelBecomesReady = new Promise((resolve, reject) => {
-            let timeout: NodeJS.Timeout;
-            if (useMaxTime) {
-              timeout = setTimeout(reject, maxTimeInSec * 1000);
-            }
-
-            self.addEventListener('model-ready', () => {
-              if (useMaxTime) {
-                clearTimeout(timeout);
-              }
-              resolve();
-            }, {once: true});
-          });
-
-          try {
-            await modelBecomesReady;
-            return null;
-          } catch (error) {
-            return `Stop capturing screenshot after ${maxTimeInSec} seconds`;
+    const evaluateError = await page.evaluate(async (maxTimeInSec) => {
+      const modelBecomesReady = new Promise((resolve, reject) => {
+        let timeout: NodeJS.Timeout;
+        if (maxTimeInSec > 0) {
+          timeout = setTimeout(() => {
+            reject(`Stop capturing screenshot after ${maxTimeInSec} seconds`);
+          }, maxTimeInSec * 1000);
+        }
+
+        self.addEventListener('model-ready', () => {
+          if (maxTimeInSec > 0) {
+            clearTimeout(timeout);
           }
-        }, useMaxTime, maxTimeInSec);
+          resolve();
+        }, {once: true});
+      });
+
+      try {
+        await modelBecomesReady;
+        return null;
+      } catch (error) {
+        return error;
```

Comment:
Not sure why the error here is a string, not an object. I guess that has something to do with puppeteer?

Reply:
I think it's because you put a string into `reject()` instead of a new Error: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject